### PR TITLE
ACMS-3447: Revisit hard dependency of entity clone module.

### PR DESCRIPTION
**Pull request by chandan-singh7929**
**Motivation**

Fixes #NNN

**Proposed changes**

**Alternatives considered**

**Testing steps**

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Review by rajeshreeputra — APPROVED** [review]

File: modules/acquia_cms_headless/src/Plugin/AcquiaCmsHeadless/HeadlessApiKeys.php [hunk 1]
```diff
@@ -227,11 +226,14 @@ public function createOperationLinks(string $entityType): array {
           ];
         }
         else {
-          $route_name = $entity->toUrl($operationLink['route'])->getRouteName();
-          $operation[$key] = [
-            'url' => Url::fromRoute($route_name, ['consumer' => $entity->id()], $destination),
-            'title' => $operationLink['title'],
-          ];
+          // Check link templete.
```
Comment (apathak18) [inline modules/acquia_cms_headless/src/Plugin/AcquiaCmsHeadless/HeadlessApiKeys.php:229]: @chandan-singh7929 Can we create separate method for this code block in the acquia_cms_headless module?

cc: @vishalkhode1

File: modules/acquia_cms_common/acquia_cms_common.module [hunk 2]
```diff
@@ -381,6 +381,19 @@ function acquia_cms_common_modules_installed($modules, $is_syncing) {
...
+
+    // Add entity_clone permission after module install.
+    if (in_array('entity_clone', $modules)) {
+      $roles = [
+        'content_administrator',
+        'content_author',
+        'content_editor',
+      ];
+      $permissionFacade = Drupal::classResolver(PermissionFacade::class);
+      foreach ($roles as $role) {
```
Comment (apathak18) [inline modules/acquia_cms_common/acquia_cms_common.module:393]: Within the same method this logic is repeated

**Review by deepakmishra2 — APPROVED** [review]
overall changes looks good except the changes mentioned by ankit. Hence moving ahead and assigning it to @apathak18

File: modules/acquia_cms_headless/src/Plugin/AcquiaCmsHeadless/HeadlessApiKeys.php (hunk 1 above)
Comment (chandan-singh7929) [reply modules/acquia_cms_headless/src/Plugin/AcquiaCmsHeadless/HeadlessApiKeys.php:229]: Created separate ticket for refactor acquia_cms_headless module : https://acquia.atlassian.net/browse/ACMS-3467

File: modules/acquia_cms_common/acquia_cms_common.module (hunk 2 above)
Comment (chandan-singh7929) [reply modules/acquia_cms_common/acquia_cms_common.module]: Done

**Review by apathak18 — APPROVED** [review]

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**